relay_messages.go

```diff
@@ -109,16 +113,19 @@ func newLazyCallReq(f *Frame) (*lazyCallReq, error) {
 
 	cr := &lazyCallReq{Frame: f}
 
-	serviceLen := f.Payload[_serviceLenIndex]
-	// nh:1 (hk~1 hv~1){nh}
-	headerStartIndex := _serviceLenIndex + 1 /* length byte */ + serviceLen
+	rbuf := typed.NewReadBuffer(f.SizedPayload()[_serviceLenIndex:])
+	rbufOffset := func() uint16 { return _serviceLenIndex + uint16(rbuf.BytesRead()) }
 
-	rbuf := typed.NewReadBuffer(f.SizedPayload()[headerStartIndex:])
+	// service~1
+	serviceLen := rbuf.ReadSingleByte()
+	rbuf.ReadBytes(int(serviceLen))
```

Comment:
We're doing a few things which we could cleanup:
 * we offset into the payload to create the rbuf
 * we then add `_serviceLenIndex` (the offset) to `BytesRead` to create the final payload aspect
 * we're calling `ReadBytes` and ignoring the result

I think these all point to a missing method in the read buffer -- a `Skip(n int)` which will skip ahead in the buffer by n bytes. That could be used to skip to the service length, and also skip the service name itself, and avoid the `rbufOffset` function completely.